stream.js

```diff
@@ -113,7 +113,10 @@ Stream.prototype._handleBlock = function(block) {
 }
 
 Stream.prototype._resume = function () {
-  if (!this.sink || this.sink.paused) return
+  if (!this.sink || this.sink.paused) {
+    this.writing = false
```

Comment:
I wonder if this line should be: `if (!this.live) this.writing = false`?

Reply:
To be safe, yeah we could do that. Then it would preserve the `live` behavior exactly as it was before.

For the future, I thought about splitting up these two implementations. There could be `live-stream.js` and `stream.js`. Would make it easier to reason about the logic in each of them.

Reply:
That is probably a good idea. At least interesting to see the diff :)